Documentation/Input/InputEvents.md

```diff
@@ -1,23 +1,169 @@
+
+1. The MRTK input system recognizes that an input event has occurred
+1. The MRTK input system fires the relevant interface function of the input event to all registered global input handlers
+1. For every active pointer registered with the input system
+    1. The input system determines which GameObject is in focus for the current pointer
+    1. The input system utilizes [Unity's Event System](https://docs.unity3d.com/Manual/EventSystem.html) to fire the relevant interface function for all matching components on the focused GameObject
+        - Example: Components implementing the interface [`IMixedRealitySpeechHandler`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealitySpeechHandler) will be searched for when a speech command is recognized
+        - Note: The Unity Event System will bubble up to search the parent GameObject if no components matching the desired interface are found on the current GameObject.
+1. If no global input handlers are registered and no GameObject is found with a matching component/interface, then the input system will call each fallback registered input handler
+
+> [!NOTE]
+> [Pointer input events](pointers.md#pointer-event-interfaces) are handled in a slightly different manner than the input event interfaces listed above. In particular, pointer input events are handled only by the GameObject in focus by the pointer that fired the input event - as well as any global input handlers. Regular input events are handled by GameObjects in focus for all active pointers.
+
+### Input event interface example
+
+The code below demonstrates use of the [`IMixedRealitySpeechHandler`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealitySpeechHandler) interface. When the user says the words "smaller" or "bigger" while focusing on a GameObject with this `ShowHideSpeechHandler` class, the GameObject will scale itself by half or twice as much.
+
+```csharp
+public class ShowHideSpeechHandler : MonoBehavior, IMixedRealitySpeechHandler
+{
+    ...
+
+    public void IMixedRealitySpeechHandler.OnSpeechKeywordRecognized(SpeechEventData eventData)
+    {
+        if (eventData.Command.Keyword == "smaller")
+        {
+            transform.localScale *= 0.5f;
+        }
+        else if (eventData.Command.Keyword == "bigger")
+        {
+            transform.localScale *= 2.0f;
+        }
+    }
+}
+```
+
+> [!NOTE]
+> [`IMixedRealitySpeechHandler`](xref:Microsoft.MixedReality.Toolkit.Input.IMixedRealitySpeechHandler) input events require that the desired keywords are pre-registered in the [MRTK Speech Commands Profile](Speech.md).
+
+## Register for global input events
+
+To create a component that listens for global input events, disregarding what GameObject may be in focus, a component must register itself with the Input System. Once registered, any instances of this MonoBehavior will receive input events along with any GameObject(s) currently in focus and other global registered listeners.
```

Comment:
It might be worth caveating that by handling an event, it will no longer be delivered to other things (i.e. if the global thing handles it, it won't get delivered to objects in focus)

Reply:
I don't believe this is true. @julenka to confirm?

Reply:
By handling, I guess I meant marking the event as used

Reply:
Ah yes ok